app/code/Magento/CatalogSearch/Model/Indexer/Fulltext/Action/DataProvider.php

```diff
@@ -561,7 +574,7 @@ private function getAttributeValue($attributeId, $valueIds, $storeId)
         if (false !== $value) {
             $optionValue = $this->getAttributeOptionValue($attributeId, $valueIds, $storeId);
             if (null === $optionValue) {
-                $value = $this->filterAttributeValue($value);
+                $value = ($value !== null) ? $this->filterAttributeValue($value) : '';
```

Comment:
Hello @karyna-tsymbal-atwix 
Please remove redundant brackets

```suggestion
                $value = $value !== null ? $this->filterAttributeValue($value) : '';
```

Reply:
done